Resources/public/js/form/edit-associations.js

```diff
@@ -0,0 +1,560 @@
+     *
+     * @typedef {{
+     *   id: string,
+     *   label: string,
+     *   editMode: string,
+     *   formType: string,
+     *   admin: string,
+     *   associationAdmin: string,
+     *   routes: {
+     *      shortObjectDescription: string,
+     *      retrieveFormElement: string,
+     *      appendFormElement: string,
+     *   }
+     * }} FieldDescriptionType
+     */
+
+    /**
+     * @typedef {{
+     *   root: jQuery,
+     *   title: jQuery,
+     *   body: jQuery,
+     *   spinner: jQuery
+     * }} DialogType
+     */
+
+    /**
+     * @typedef {{
+     *   fieldDescription: FieldDescriptionType,
+     *   dialog: ?DialogType
+     * }} FieldActionType
+     */
+
+    //
+    // Utilities
+    //-----------------------------------------------------------------------------------------------------------------
+
+    function merge () {
+        return $.extend.apply(null, [].concat.apply([true, {}], arguments));
+    }
+
+    /**
+     * Returns whether the given element is an anchor inside the same page.
+     *
+     * @param {jQuery} $el
+     * @returns {boolean}
+     */
+    function isAnchor ($el) {
+        var href = $el.attr('href');
+        return $el.is('a') && (!href || href[0] === '#');
+    }
+
+    /**
+     * Retrieves the form field description from a DOM Element.
+     * The field description must be passed to the element as a serialized JSON object,
+     * via a [data-field-description] attribute.
+     * The JSON object must conform to the {FieldDescriptionType} definition.
+     *
+     * @param {jQuery} $element
+     * @returns {FieldDescriptionType}
+     */
+    function getFieldDescription ($element) {
+        var fieldDescription = $element.data('fieldDescription');
+        fieldDescription.actionType = $element.data('fieldAction');
+        return fieldDescription;
+    }
+
+    // ==================== Former edit_many_association_script.html.twig ==================== //
+
+    //
+    // Dialog
+    //-----------------------------------------------------------------------------------------------------------------
+
+    var DIALOG_TEMPLATE =
+        '<div class="modal fade" tabindex="-1" role="dialog" aria-hidden="true">' +
+            '<div class="modal-dialog modal-lg">' +
+                '<div class="modal-content">' +
+                    '<div class="modal-header">' +
+                        '<button class="close" data-dismiss="modal" aria-hidden="true" aria-label="Close">' +
+                            '<i class="fa fa-close"></i>' +
+                        '</button>' +
+                        '<h4 class="modal-title"></h4>' +
+                    '</div>' +
+                    '<div class="modal-body">' +
+                        '<div class="modal-body-content"></div>' +
+                        '<div class="overlay spinner-overlay text-primary"></div>' +
+                    '</div>' +
+                '</div>' +
+            '</div>' +
+        '</div>'
+    ;
```

Comment:
Honestly, there's absolutely no difference between this and
```js
$('<div class="modal">')
  .append('<div class="modal-content">')
    .append('<div class="modal-header">')
      .append('<button class="close">').attr('aria-hidden', true)
// etc...
```
Except that the first is shorter and more readable...

> Having html in .js does not feel very good…

But it's gonna happen all the time because **that's** what JS is for ! :wink: